include/boost/beast/core/impl/buffers_adaptor.hpp

```diff
@@ -601,6 +437,329 @@ consume(std::size_t n) noexcept
     }
 }
 
+template<class MutableBufferSequence>
+auto
+buffers_adaptor<MutableBufferSequence>::
+make_subrange(std::size_t pos, std::size_t n)
+-> subrange<true>
+{
+    return subrange<true>(
+        begin_, net::buffer_sequence_end(bs_),
+        in_pos_ + pos, n);
+}
+
+template<class MutableBufferSequence>
+auto
+buffers_adaptor<MutableBufferSequence>::
+make_subrange(std::size_t pos, std::size_t n) const
+-> subrange<false>
+{
+    return subrange<false>(
+        begin_, net::buffer_sequence_end(bs_),
+        in_pos_ + pos, n);
+}
+
+// -------------------------------------------------------------------------
+// subrange
+
+template<class MutableBufferSequence>
+template<bool isMutable>
+buffers_adaptor<MutableBufferSequence>::
+subrange<isMutable>::
+subrange(
+    iter_type first,  // iterator to first buffer in storage
+    iter_type last,   // iterator to last buffer in storage
+    std::size_t pos,        // the offset in bytes from the beginning of the storage
+    std::size_t n)         // the total length of the subrange
+    : first_(first)
+    , last_(last)
+    , first_offset_(0)
+    , last_size_((std::numeric_limits<std::size_t>::max)())
+{
+    adjust(pos, n);
+}
+
+template<class MutableBufferSequence>
+template<bool isMutable>
+auto
+buffers_adaptor<MutableBufferSequence>::
+subrange<isMutable>::
+adjust(
+    std::size_t pos,
+    std::size_t n)
+-> void
+{
+    if (n == 0)
+        last_ = first_;
+
+    // note: fallthrough
+
+    if (first_ == last_)
+    {
+        first_offset_ = 0;
+        last_size_ = 0;
+        return;
+    }
+
+    auto last_element = [this](iter_type iter) {
+        return std::next(iter) == last_;
+    };
+
+
+    pos += first_offset_;
+    while (pos)
+    {
+        auto adjust = (std::min)(pos, first_->size());
+        if (adjust >= first_->size())
+        {
+            ++first_;
+            first_offset_ = 0;
+            pos -= adjust;
+        }
+        else
+        {
+            first_offset_ = adjust;
+            pos = 0;
```

Comment:
you could `break` and skip the loop condition...

Reply:
This would have no effect on the quality of gcc's code, and would for me make the loop more difficult to reason about as it's another exit point.

https://godbolt.org/z/yNsgjF

nevertheless will add